### DISCUSSION TITLE
Dependency not work when using with APIRouter

### DISCUSSION THREAD

**Discussion by johnsxiongs** [fastapi-users/fastapi-users#456] [Q&A]
the tutorial  example here well worked
```
@app.get("/protected-route")
def protected_route(user: User = Depends(fastapi_users.get_current_user)):
      return f"Hello, {user.email}"
```

My code with FastAPI Utilities  not work 
```
from fastapi import APIRouter
router = APIRouter()

@router.get("/protected-route")
def protected_route(user: User = Depends(fastapi_users.get_current_user)):
      return f"Hello, {user.email}"
```
Wrong response :
![image](https://user-images.githubusercontent.com/37213061/105574804-f404b880-5da1-11eb-97c2-11ca41a90e4d.png)

**Comment by frankie567** [comment]
Hi @johnxiongs!

Sorry for the response delay!

The problem here is that you define your route under the `/users` routes. The route `/users/protected-route` is actually catched by the FastAPI Users route to retrieve a single user (https://frankie567.github.io/fastapi-users/usage/routes/#get-user_id) ; that requires a superuser (hence the 403 error).

There is an ongoing discussion about this (#444), that should help with this. In the meantime, I suggest you define your route with another path.
